optimum_benchmark/benchmarks/training.py

```diff
@@ -46,193 +45,58 @@ class TrainingConfig(BenchmarkConfig):
     def __init__(self):
         # initialize training results
         self.training_metrics: Dict[str, Any] = {}
 
     def configure(self, config: TrainingConfig):
         super().configure(config)
 
-        self.dataset_shapes = config.dataset_shapes
-        self.training_arguments = config.training_arguments
-
     def run(self, backend: "Backend") -> None:
         LOGGER.info("Running training benchmark")
-        model_shapes = backend.model_shapes
-        self.dataset_shapes = {**self.dataset_shapes, **model_shapes}
-
-        self.dataset_generator = DatasetGenerator(
-            task=backend.task,
-            dataset_shapes=self.dataset_shapes,
+        task = backend.task
+        dataset_shapes = {**self.config.dataset_shapes, **backend.model_shapes}
+        dataset_generator = DatasetGenerator(task=task, dataset_shapes=dataset_shapes)
+
+        training_dataset = dataset_generator.generate()
+        training_data_collator = get_data_collator(task=task)
+        training_callbacks = [MeasurementCallback(self.config.warmup_steps)]
+
+        trainer_state = backend.train(
+            training_dataset=training_dataset,
+            training_callbacks=training_callbacks,
+            training_data_collator=training_data_collator,
+            training_arguments=self.config.training_arguments,
         )
 
-        training_dataset = self.dataset_generator.generate()
-
-        training_data_collator = get_data_collator(
-            task=backend.task,
-        )
-
-        if backend.config.name == "pytorch":
-            self.training_metrics = backend.run_pytorch_training(
-                training_config=self.config,
-                training_arguments=self.training_arguments,
-                training_dataset=training_dataset,
-                training_data_collator=training_data_collator,
-            )
-        else:
-            backend.prepare_for_training(
-                training_dataset=training_dataset,
-                training_data_collator=training_data_collator,
-                training_arguments=self.training_arguments,
-            )
-            training_output = backend.train()
-
-            self.training_metrics = {
-                "training_throughput": training_output.metrics[
-                    "train_samples_per_second"
-                ],
-                "train_runtime": training_output.metrics["train_runtime"],
-            }
+        self.training_metrics = {
+            # warmup metrics
+            "warmup_runtime": trainer_state.warmup_runtime,
+            "warmup_throughput()": trainer_state.warmup_samples_per_second,
+            # training metrics
+            "train_runtime": trainer_state.train_runtime,
+            "training_throughput": trainer_state.train_samples_per_second,
+            # overall training metrics
+            "overall_train_runtime": trainer_state.overall_train_runtime,
+            "overall_training_throughput": trainer_state.overall_train_samples_per_second,
```

Comment:
should probably align on train or training

Reply:
Yes